library/src/main/scala/scala/scalajs/LinkingInfo.scala

```diff
@@ -108,4 +108,35 @@ object LinkingInfo {
   def assumingES6: Boolean =
     linkingInfo.assumingES6
 
+  /** The value of the global JavaScript `this`.
+   *
+   *  This returns the value that would be obtained by writing `this` at the
+   *  top-level of the JavaScript file generated by Scala.js. In most
+   *  JavaScript environments, this is equivalent to the *global object*, but
+   *  it is not necessarily the case. For example, on Node.js, `this` is the
+   *  object representing the `exports` of the current module.
+   *
+   *  Using this value should be rare, and mostly limited to writing code
+   *  detecting what the global object is. For example, a typical detection
+   *  code looks like:
+   *  {{{
+   *  val globalObject = {
+   *    import js.Dynamic.{global => g}
+   *    if (js.typeOf(g.global) != "undefined" && (g.global.Object eq g.Object)) {
+   *      // Node.js environment detected
+   *      g.global
+   *    } else {
+   *      // In all other well-known environment, we can use the global `this`
+   *      scala.scalajs.LinkingInfo.globalThis
+   *    }
+   *  }
+   *  }}}
+   *  Note that the above code is not comprehensive, as there can be JavaScript
+   *  environments where the global object cannot be fetched neither through
+   *  `global` nor `this`. If your code needs to run in such an environment, it
+   *  is up to you to use an appropriate detection procedure.
+   */
+  @inline
+  def globalThis: js.Dynamic = linkingInfo.globalThis
+
```

Comment:
Do you object to it being defined in `scalajs.runtime.linkingInfo.globalThis` too, or only in `scalajs.LinkingInfo`? If it can't be in `scalajs.runtime.linkingInfo.globalThis`, then it needs to be yet another primitive with its own IR node. That's only the private API though, so I don't think it's a significant issue.

For the public API:

* `js.Dynamic.globalThis`: that was my first go-to, but I don't like it because it gives too much "visibility" to that method. It should almost never be used. Putting it next to high-profile methods such as `js.Dynamic.global` and `js.Dynamic.literal` feels wrong.
* `js.globalThis`: similar argument. I'm not a big fan of showing it in such a visible part of the API.
* `js.EnvInfo`: I have secret plans to kill `envInfo` altogether. We won't need it now that a) `global` is not there anymore, b) `exportsNamespace` is dying and c) the other things in there can be accessed with `global.__ScalaJSEnv`.

If I had to choose between one of those three, I'd take `js.globalThis`.